vms/evm/predicate/bytes_test.go

```diff
@@ -0,0 +1,339 @@
+			expected: []byte{0x42, 0xff},
+		},
+		{
+			name:     "exactly 31 bytes",
+			input:    bytes.Repeat([]byte{0xaa}, 31),
+			expected: append(bytes.Repeat([]byte{0xaa}, 31), 0xff),
+		},
+		{
+			name:     "exactly 32 bytes",
+			input:    bytes.Repeat([]byte{0xbb}, 32),
+			expected: append(bytes.Repeat([]byte{0xbb}, 32), 0xff),
+		},
+		{
+			name:     "exactly 63 bytes",
+			input:    bytes.Repeat([]byte{0xcc}, 63),
+			expected: append(bytes.Repeat([]byte{0xcc}, 63), 0xff),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			packed := New(tt.input)
+			unpacked, err := Unpack(packed)
+
+			require.NoError(err)
+			require.Equal(tt.input, unpacked)
+
+			// Verify the packed result has the correct structure
+			require.Equal(tt.expected, []byte(packed)[:len(tt.expected)])
+
+			// Verify padding is all zeros
+			for i := len(tt.expected); i < len(packed); i++ {
+				require.Equal(byte(0), packed[i])
+			}
+		})
+	}
+}
+
+func TestPackUnpackPadding(t *testing.T) {
+	require := require.New(t)
+
+	// Test that padding is correctly applied to reach multiple of 32
+	testCases := []int{0, 1, 31, 32, 33, 63, 64, 65}
+
+	for _, length := range testCases {
+		t.Run(fmt.Sprintf("length_%d", length), func(_ *testing.T) {
+			input := utils.RandomBytes(length)
+			packed := New(input)
+			unpacked, err := Unpack(packed)
+
+			require.NoError(err)
+			require.Equal(input, unpacked)
+
+			// Verify packed length is a multiple of 32
+			require.Equal(0, len(packed)%32)
+		})
+	}
+}
+
+func TestUnpackInvalid(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       Predicate
+		expectedErr error
+	}{
+		{
+			name:        "empty input",
+			input:       Predicate{},
+			expectedErr: errInvalidEmptyPredicate,
+		},
+		{
+			name:        "all zeros",
+			input:       Predicate(bytes.Repeat([]byte{0}, 32)),
+			expectedErr: errInvalidAllZeroBytes,
+		},
+		{
+			name:        "missing delimiter",
+			input:       Predicate(bytes.Repeat([]byte{0x42}, 32)),
+			expectedErr: errInvalidEndDelimiter,
+		},
+		{
+			name:        "wrong delimiter",
+			input:       Predicate(append(bytes.Repeat([]byte{0x42}, 31), 0x00)),
+			expectedErr: errInvalidEndDelimiter,
+		},
+		{
+			name:        "excess padding",
+			input:       Predicate(append(append([]byte{0x42, 0xff}, bytes.Repeat([]byte{0}, 30)...), 0x01)),
+			expectedErr: errInvalidPadding,
+		},
+		{
+			name:        "non-zero padding",
+			input:       Predicate(append(append([]byte{0x42, 0xff}, bytes.Repeat([]byte{0}, 29)...), 0x01, 0x00)),
+			expectedErr: errInvalidPadding,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			_, err := Unpack(tt.input)
+			require.ErrorIs(err, tt.expectedErr)
+		})
+	}
+}
+
+func FuzzPackUnpackRoundTrip(f *testing.F) {
+	// Seed with various input sizes
+	for i := range 100 {
+		f.Add(utils.RandomBytes(i))
+	}
+
```

Comment:
This does not meaningfully improve the fuzzer's searching (using a default corpus is fine if we don't have anything good to populate it with.
```suggestion
```

Alternatively, we could seed it with valid predicates.